graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java

```diff
@@ -91,15 +99,36 @@ public int getOrder() {
         return POSITION; // allow mutation of metadata
     }
 
+    @Override
+    public Set<String> getSupportedAnnotationNames() {
+        return CollectionUtils.setOf(
+                ReflectiveAccess.class.getName(),
+                TypeHint.class.getName(),
+                Import.class.getName()
+        );
+    }
+
+    @NonNull
+    @Override
+    public VisitorKind getVisitorKind() {
+        return VisitorKind.AGGREGATING;
+    }
+
     @Override
     public void visitClass(ClassElement element, VisitorContext context) {
         if (!isSubclass && !element.hasStereotype(Deprecated.class)) {
             if (element.hasAnnotation(ReflectiveAccess.class)) {
+                originatingElements.add(element);
                 packages.add(element.getPackageName());
                 final String beanName = element.getName();
                 addBean(beanName);
-                resolveClassData(beanName + "[]");
+                if (element.isArray()) {
+                    resolveClassData(beanName + "[]");
+                } else {
+                    resolveClassData(beanName);
+                }
```

Comment:
@ilopmar this appeared to be a bug? Please confirm

Reply:
Someone submitted a PR sometime ago to also include arrays in the reflection config for all the declared classes. Apparently they needed for their use case